File: api/src/main/java/club/yunzhi/smartcommunity/entity/Duty.java
package club.yunzhi.smartcommunity.entity;

import org.hibernate.annotations.SQLDelete;
;

import javax.persistence.Column;
import javax.persistence.Entity;

/**
 * 职务实体
 */
@Entity
@SQLDelete(sql = "update `duty` set deleted = 1 where id = ?")
public class Duty extends SoftDeleteEntity {

  public static String POST_TOWN = "town";
  public static String POST_COMMUNITY = "community";

  /**
   * 名称
   */
  @Column(nullable = false)
  private String name;

  /**
   * 权重
   */
  @Column(nullable = false)
  private Long weight;

  /**
   * 区域类型
   */
  @Column(nullable = false)
  private String typeOfDistrict;

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public Long getWeight() {
    return weight;
  }

  public void setWeight(Long weight) {
    this.weight = weight;
  }

  public String getTypeOfDistrict() {
    return typeOfDistrict;
  }

  public void setTypeOfDistrict(String typeOfDistrict) {
    this.typeOfDistrict = typeOfDistrict;
  }
}

File: web/projects/lib/src/entity/duty.ts
/**
 * 岗位
 */
import {DistrictType} from "./enum/district-type";

export class Duty {
  /**
   * id
   */
  id: number;

  /**
   * 名称
   */
  name: string;

  /**
   * 权重
   */
  weight: number;

  /**
   * 区域类型
   */
  typeOfDistrict: DistrictType;

  constructor(data = {} as {
    id?: number,
    name?: string,
    weight?: number,
    typeOfDistrict?: DistrictType
  }) {
    if (data) {
      this.id = data.id;
      this.name = data.name;
      this.weight = data.weight;
      this.typeOfDistrict = data.typeOfDistrict;
    }
  }
}
